libraries/AP_BattMonitor/AP_BattMonitor_MAV.cpp

```diff
@@ -0,0 +1,108 @@
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+#include "AP_BattMonitor_MAV.h"
+
+AP_BattMonitor_MAVLink_Battery::AP_BattMonitor_MAVLink_Battery(AP_BattMonitor &mon,
+                                             AP_BattMonitor::BattMonitor_State &mon_state,
+                                             AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+	_state.healthy = false;
+}
+
+void AP_BattMonitor_MAVLink_Battery::read()
+{
+    if ((AP_HAL::micros() - _state.last_time_micros) > MAVLINK_SMART_BATTERY_TIMEOUT_MICROS) {
+		_state.healthy = false;
+		return;
+	} 
+	_state.healthy = true;	
+   return;
+}
+
+bool AP_BattMonitor_MAVLink_Battery::capacity_remaining_pct(uint8_t &cap_remain_pct) const
+{
+    if (_capacity_rem_pct == INT8_MIN) {
+		return false;
+	}
+	/* positive capacity remaining is expected */
+	if (_capacity_rem_pct > 0) {
+		cap_remain_pct = _capacity_rem_pct;
+	} else {
+		cap_remain_pct = 0;
+	}
+	return true;
+}
+
+bool AP_BattMonitor_MAVLink_Battery::get_cycle_count(uint16_t &cycles) const
+{
+    if (_cycle_count != UINT16_MAX) {
+		cycles = _cycle_count;
+    	return true;
+	}
+	return false;
+}
+
+void AP_BattMonitor_MAVLink_Battery::process_info_msg(const mavlink_message_t &msg)
+{
+	int32_t capacity = mavlink_msg_smart_battery_info_get_capacity_full(&msg);
+	if (capacity != -1) {

```

Comment:
Why not just decode the pecket here like you do the other packet? I know there's lots of unused bytes, but it would be easier to read and consistent with all other drivers. This is a low frequency packet, right?